packages/core/src/tracing/clsHooked/async_local_storage_context.js

```diff
@@ -0,0 +1,324 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+/**
+ * This is a reimplementation of the cls-hooked API on top of AsyncLocalStorage.
+ * This implementation is used instead of the cls-hooked based implementation in Node.js runtimes where
+ * AsyncLocalStorage is available. It exposes the same API as the previous implementation, and can be used
+ * interchangeably.
+ */
+
+'use strict';
+
+const util = require('util');
+const assert = require('assert');
+const wrapEmitter = require('./emitter-listener');
+const { AsyncLocalStorage, executionAsyncId } = require('async_hooks');
+const unset = require('./unset');
+
+const CONTEXTS_SYMBOL = 'instanaClsHooked@contexts';
+
+const storage = new AsyncLocalStorage();
+
+module.exports = {
+  createNamespace,
+  reset,
+  destroyNamespace,
+  getNamespace
+};
+
+/**
+ * @typedef {Object.<string, *>} InstanaCLSContext
+ */
+
+/**
+ * Creates a new CLS namespace.
+ */
+class Namespace {
+  /**
+   * @param {string} name
+   */
+  constructor(name) {
+    /**
+     * The idea behind sharedContext is to keep a reference of a finished context - in normal circumstances - but that
+     * we wish to use it as a parent context for the next instrumentation.
+     * For instance, we need a shared context in the GraphQL instrumentation when HTTP is used as a client entry point.
+     * In this case, the HTTP server context is properly managed by the Async Localstorage and the context is exited by
+     * it. But we need this context as the parent of the GraphQL instrumentation.
+     */
+    this.sharedContext = null;
+    this.name = name;
+    this.active = null;
+    this.id = -1;
+    this._contexts = new Map();
```

Comment:
I guess we can remove `this._contexts` as it was replaced by async local storage keeping track of the context.

Reply:
I agree, that seems to be unused.